### ISSUE TITLE
Unhandled error on connection

### ISSUE THREAD

**Issue by glenarama** [Bugs5382/node-hl7-client#72]
Thanks for the library!

Receiving an error on connection - any help appreciated:
Version: node-hl7-client 1.1.3

```
try {
  const client = new Client({ host: "xx.xx.xx.xx" });
  client.createOutbound({ port: xxxxx }, async () => {});
} catch (err) {
  console.log("Error", err);
}
```

Client creation seems fine, but outbound request throws:

```
  Error [ERR_UNHANDLED_ERROR]: Unhandled error. (undefined)
  at new NodeError (node:internal/errors:399:5)
  at HL7Outbound.emit (node:events:501:17)
  at Socket.eval (webpack-internal:///(rsc)/../../node_modules/.pnpm/node-hl7-client@1.1.3/node_modules/node-hl7-client/lib/esm/client/hl7Outbound.js:259:26)
  at Socket.emit (node:events:512:28)
  at TCP.<anonymous> (node:net:332:12)
  at TCP.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'ERR_UNHANDLED_ERROR',
  context: undefined
}
```

**Comment by Bugs5382** [comment]
Anything else in your code? Or Just that?

**Comment by glenarama** [comment]
Thanks for looking at this:


> Anything else in your code? Or Just that?

Just the import - full index.ts file: 

```
import { Client } from "node-hl7-client";

try {
  const client = new Client({ host: "xx.xx.xx.xx" });
  client.createOutbound({ port: xxxxx }, async () => {});
} catch (err) {
  console.log("Error", err);
}
```

**Comment by Bugs5382** [comment]
Roger. I just ran it @ work real quick. Yep. An error. *sigh*. Ok will do a deep dive tonight.

**Comment by glenarama** [comment]
Am sending to a third party - it is a valid destination but i can't confirm if their server is responding correctly.

**Comment by Bugs5382** [comment]
Ok. You and I are having the same issue then.... 

So two errors:

- [ ] Fix Unhandled error (Found a Fix)
- [ ] Unable to connect to 3rd party

**Comment by Bugs5382** [comment]
So.. more updates..

```ts
const {Client, Message} = require("node-hl7-client");

try {
    const client = new Client({ host: 'x.x.x.x'})
    const test = client.createOutbound({port: 19999 }, async (res) => {
        const message = res.getMessage()
        console.log(message.toString())
    })
    const message = new Message({
        messageHeader: {
            msh_9_1: "ADT",
            msh_9_2: "A01",
            msh_10: "1234567890",
            msh_11_1: "D"
        }
    })
    test.sendMessage(message)
    test.close()
} catch (e) {
    console.log(e)
}
```

When I did this and did a wireshark capture on my Corepoint server, I saw:

![image](https://github.com/Bugs5382/node-hl7-client/assets/12115015/a2236140-19b3-4b53-8f34-857f8597be3a)

And Corepoint was processing it correctly (IB on corepoint is not set to send the ACK right now), but it did process it,

Maybe it's a design issue...

```createOutbound``` doesn't try to "connect" until sendMessage is done and then doing the "open" and "close" in that seqeuence making the code simpler (and breaking the API).

**Comment by Bugs5382** [comment]
I am working on revising the structure so that sendMessage is the only thing being used. More than likely I will mark this breaking. I am not sure how many people are using this NPM package yet...

**Comment by Bugs5382** [comment]
Almost got this corrected...

**Comment by glenarama** [comment]
Great! Let me know if you'd like some testing on a version.

**Comment by Bugs5382** [comment]
I haven't commit the bulk changes even locally yet because I just want to make sure that I am tracking the changes. It will break a lot of the current API so I am being careful. I am checking all my other unit testing outside of this internal checking so it's a lot longer process. Tried to get it out this weekend, but just to much to look over. Needs to be perfect.

**Comment by Bugs5382** [comment]
@glenarama the code is pushed. still trying to fix a linting issue, but my head to to tired for get to it tonight.

**Comment by Bugs5382** [comment]
@glenarama  Ok.. so release..

Here is a quick script:
```
const client = new Client({ host: 'x.x.x.x' })
const outbound = client.createConnection({ port: 19999 }, async (res) => {
    const messageRes = res.getMessage()
    console.log(messageRes.toString())
})
let message = new Message({
    messageHeader: {
        msh_9_1: 'ADT',
        msh_9_2: 'A01',
        msh_10: 'CONTROL_ID',
        msh_11_1: 'D'
    }
})
outbound.sendMessage(message)
```

that I been using as my test to a Corepoint server vs. my node-hl7-server package. I tested it out there too, but I have to update that package to work with this 2.0.0-beta.1  before fully committing this to main.

**Comment by glenarama** [comment]
I've managed to get a working connection with the new release Thankyou!. 
A couple of question i can't find documented or in the code:
- Im getting no timeout response, is there a way to handle timeout - it looks like the connection remains pending.
- When sending rejected / invalid messages i'm getting no error response

**Comment by Bugs5382** [comment]
@glenarama A fix in 2.3.1 fixes this issue.
